core/src/main/scala/org/bitcoins/core/protocol/blockchain/ChainParams.scala

```diff
@@ -104,6 +100,52 @@ sealed abstract class ChainParams {
                        amount)
   }
 
+  /**
+   * Creates the Genesis [[org.bitcoins.core.protocol.blockchain.Block Block]] for Signet.
+   *
+   * @see Mimics
+   *      [[https://github.com/bitcoin/bitcoin/blob/006d9ab9e59a00fa02b8d5f32919f9d57567fa65/src/chainparams.cpp#L67 this function]]
+   *      in Bitcoin Core
+   *
+   * @param time the time when the miner started hashing the block header
+   * @param nonce the nonce to mine the block
+   * @param nBits An encoded version of the target threshold this block’s header hash must be less than or equal to.
+   * @param version the block version
+   * @param amount the block reward for the genesis block (50 BTC in Bitcoin)
+   * @return the newly minted genesis block
+   */
+  def createSignetGenesisBlock(
+      time: UInt32,
+      nonce: UInt32,
+      nBits: UInt32,
+      version: Int32,
+      amount: CurrencyUnit): Block = {
+    val defaultSignetKey = DoubleSha256Digest(
+      "03ad5e0edad18cb1f0fc0d28a3d4f1f3e445640337489abb10404f2d1e086be430")
+    val const = ScriptConstant(defaultSignetKey.bytes)
+
+    val asm = {
+      List(OP_1, const, OP_1, OP_CHECKMULTISIG)
+    }
+
+    val blockScript = ScriptSignature.fromAsm(asm)
```

Comment:
i'm not sure how signet's signing works, but you might want to use our built in script types.

You can create a multisig script with

```scala
scala> import org.bitcoins.core.protocol.script._
import org.bitcoins.core.protocol.script._

scala> import org.bitcoins.core.crypto._
import org.bitcoins.core.crypto._

scala> val pubKey = ECPublicKey.fromHex("03ad5e0edad18cb1f0fc0d28a3d4f1f3e445640337489abb10404f2d1e086be430")
pubKey: org.bitcoins.core.crypto.ECPublicKey = ECPublicKey(03ad5e0edad18cb1f0fc0d28a3d4f1f3e445640337489abb10404f2d1e086be430)

scala> val m = MultiSignatureScriptPubKey(1,Vector(pubKey))
m: org.bitcoins.core.protocol.script.MultiSignatureScriptPubKey = MultiSignatureScriptPubKey(1, 1, ECPublicKey(03ad5e0edad18cb1f0fc0d28a3d4f1f3e445640337489abb10404f2d1e086be430))

scala> m.asm
res0: Seq[org.bitcoins.core.script.constant.ScriptToken] = Vector(OP_1, BytesToPushOntoStackImpl(33), ScriptConstantImpl(ByteVector(33 bytes, 0x03ad5e0edad18cb1f0fc0d28a3d4f1f3e445640337489abb10404f2d1e086be430)), OP_1, OP_CHECKMULTISIG)
```

This also makes sure you get the asm format correct :-).

Depending on how signet works (raw multisig, vs wrapped in p2wsh/p2sh) you can wrap that script like this:

```scala
scala> val p2sh = P2SHScriptPubKey(m)
p2sh: org.bitcoins.core.protocol.script.P2SHScriptPubKey = P2SHScriptPubKey(Sha256Hash160DigestImpl(04699beafffcd5b24d48e5b5a9f02979aa92d2c0))

scala> val p2wsh = P2WSHWitnessSPKV0(m)
p2wsh: org.bitcoins.core.protocol.script.P2WSHWitnessSPKV0 = wsh(Sha256DigestImpl(d64e78dc1940a14c251ab244ee7d54999018e72894ae85d8d7baf710be6215b6))

```

Reply:
Signet is raw, I believe; no wrappery. (In fact, I don't think it supports segwit right now. Probably worth looking at, but not hugely important.)